aws/resource_aws_instance_test.go

```diff
@@ -4659,3 +4545,78 @@ resource "aws_instance" "test" {
 }
 `
 }
+
+// testAccLatestAmazonLinuxHvmEbsAmiConfig returns the configuration for a data source that
+// describes the latest Amazon Linux AMI using HVM virtualization and an EBS root device.
+// The data source is named 'amzn-ami-minimal-hvm-ebs'.
+func testAccLatestAmazonLinuxHvmEbsAmiConfig() string {
+	return fmt.Sprintf(`
+data "aws_ami" "amzn-ami-minimal-hvm-ebs" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name = "name"
+    values = ["amzn-ami-minimal-hvm-*"]
+  }
+
+  filter {
+    name = "root-device-type"
+    values = ["ebs"]
+  }
+}
+`)
+}
+
+// testAccLatestAmazonLinuxPvEbsAmiConfig returns the configuration for a data source that
+// describes the latest Amazon Linux AMI using PV virtualization and an EBS root device.
+// The data source is named 'amzn-ami-minimal-pv-ebs'.
+/*
+func testAccLatestAmazonLinuxPvEbsAmiConfig() string {
+	return fmt.Sprintf(`
+data "aws_ami" "amzn-ami-minimal-hvm-ebs" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name = "name"
+    values = ["amzn-ami-minimal-pv-*"]
+  }
+
+  filter {
+    name = "root-device-type"
+    values = ["ebs"]
+  }
+}
+`)
+}
+*/
+
+// testAccAwsInstanceVpcConfig returns the configuration for resources that create a VPC and subnet in that the VPC.
+// The resources are named 'test'.
+func testAccAwsInstanceVpcConfig(rName string) string {
+	return fmt.Sprintf(`
+data "aws_availability_zones" "current" {
+  # Exclude us-west-2d as it has limited instance types.
+  blacklisted_names = ["us-west-2d"]
```

Comment:
This will actually want to be `blacklisted_zone_ids` instead of `blacklisted_names` for consistency (more info on why [here](https://docs.aws.amazon.com/ram/latest/userguide/working-with-az-ids.html)).